inc/msp/message.hpp

```diff
@@ -0,0 +1,67 @@
+#ifndef MESSAGE_HPP
+#define MESSAGE_HPP
+
+#include "variants.hpp"
+#include "msp_id.hpp"
+#include "byte_vector.hpp"
+
+#include <vector>
+#include <string>
+#include <memory>
+#include <iostream>
+
+
+
+
+namespace msp {
+
+
+class Message {
+public:
+    virtual ID id() const = 0;
+    
+    Message() {};
```

Comment:
You can remove this default constructor and just set a default value for the constructor that takes the firmware type.